pgjdbc/src/main/java/org/postgresql/util/ExpressionProperties.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2017, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import java.util.Properties;
+
+public class ExpressionProperties extends Properties {
+
+  /**
+   * Prefix for constant names
+   */
+  private static final String PREFIX = "${";
+
+  /**
+   * Suffix for constant names
+   */
+  private static final String SUFFIX = "}";
+
+  /**
+   * Creates an empty property list with no default values.
+   */
+  public ExpressionProperties() {
+    super();
+  }
+
+  /**
+   * Creates an empty property list with the specified defaults.
+   *
+   * @param defaults java.util.Properties
+   */
+  public ExpressionProperties(Properties defaults) {
+    super(defaults);
+  }
+
+  /**
+   * This method expands the constants between the PREFIX and SUFFIX and return the normalized value.
+   *
+   * The method returns <code>null</code> if the property is not found.
+   *
+   * @param key the property key.
+   *
+   * @return the value in this property list with
+   *         the specified key value.
+   */
+  public String getProperty(String key) {
+    String value = super.getProperty(key);
+    if (value != null) {
```

Comment:
I would prefer early exit in such cases.
E.g. `if (value == null) { return null; }`